packages/transport/src/core/Reference.ts

```diff
@@ -29,14 +32,22 @@ export class Reference extends Serializable implements IReference {
     @serialize()
     public forIdentityTruncated?: string;
 
-    @validate({ nullable: true, min: 1, max: 12, customValidator: (v) => (!Number.isInteger(v) ? "must be an integer" : undefined) })
+    @validate({ nullable: true, regExp: /^(pw|pin(4|5|6|7|8|9|10|11|12|13|14|15|16))$/ })
     @serialize()
-    public passwordType?: number;
+    public passwordType?: string;
+
+    @validate({ nullable: true, min: 1, customValidator: (v: number) => (Number.isInteger(v) ? undefined : "must be an integer") })
+    @serialize()
+    public version?: number;
+
+    @validate({ nullable: true, customValidator: (v: ICoreBuffer) => (v.buffer.byteLength === 16 ? undefined : "must be 16 bytes long") })
+    @serialize()
+    public salt?: CoreBuffer;
 
     public truncate(): string {
         const idPart = this.backboneBaseUrl ? `${this.id.toString()}@${this.backboneBaseUrl}` : this.id.toString();
         const truncatedReference = CoreBuffer.fromUtf8(
-            `${idPart}|${this.key.algorithm}|${this.key.secretKey.toBase64URL()}|${this.forIdentityTruncated ? this.forIdentityTruncated : ""}|${this.passwordType ? this.passwordType.toString() : ""}`
+            `${idPart}|${this.key.algorithm}|${this.key.secretKey.toBase64URL()}|${this.forIdentityTruncated ? this.forIdentityTruncated : ""}|${this.version ? this.version : ""}|${this.salt ? this.salt.toBase64() : ""}|${this.passwordType ? this.passwordType : ""}`
```

Comment:
now you introduced something we declared as a **definite** non-goal, having a 3,5,7 length. I remember that we talked about to introduce 3 for backwards compat, but all new features have to be empty or filled, so why 5 or 7 instead of just 7?

Reply:
What do we do if someone has created correct References with an older length (in this case 5)?

Reply:
Also, IMO we can archive to bring this back to 5, but that's a next step.

Reply:
misspelled. I meant generate.